pxr/imaging/plugin/hdEmbree/ies.cpp

```diff
@@ -0,0 +1,490 @@
+    if (ies.empty()) {
+        return false;
+    }
+
+    IESTextParser parser(ies);
+    if (parser.eof()) {
+        return false;
+    }
+
+    // Handle the tilt data block
+    if (strncmp(parser.data, "\nTILT=INCLUDE", 13) == 0) {
+        parser.data += 13;
+        parser.get_double();              //< Lamp to Luminaire geometry
+        int num_tilt = parser.get_long(); //< Amount of tilt angles and factors
+
+        // Skip over angles and factors
+        for (int i = 0; i < 2 * num_tilt; i++) {
+            parser.get_double();
+        }
+    } else {
+        // Skip to next line
+        parser.data = strstr(parser.data + 1, "\n");
+    }
+
+    if (parser.eof()) {
+        return false;
+    }
+    parser.data++;
+
+    parser.get_long();                    //< Number of lamps
+    parser.get_double();                  //< Lumens per lamp
+    double factor = parser.get_double();  //< Candela multiplier
+    int v_angles_num = parser.get_long(); //< Number of vertical angles
+    int h_angles_num = parser.get_long(); //< Number of horizontal angles 
+    _type = (IESType)parser.get_long();   //< Photometric type
+
+    // TODO(lukas): Test whether the current type B processing can also deal with
+    // type A files. In theory the only difference should be orientation which we
+    // ignore anyways, but with IES you never know...
+    //
+    if (_type != TYPE_B && _type != TYPE_C) {
+        return false;
+    }
+
+    parser.get_long();             //< Unit of the geometry data
+    parser.get_double();           //< Width
+    parser.get_double();           //< Length
+    parser.get_double();           //< Height
+    factor *= parser.get_double(); //< Ballast factor
+    factor *= parser.get_double(); //< Ballast-Lamp Photometric factor
+    parser.get_double();           //< Input Watts
+
+    _v_angles.reserve(v_angles_num);
+    for (int i = 0; i < v_angles_num; i++) {
+        _v_angles.push_back((float)parser.get_double());
+    }
+
+    _h_angles.reserve(h_angles_num);
+    for (int i = 0; i < h_angles_num; i++) {
+        _h_angles.push_back((float)parser.get_double());
+    }
+
+    _intensities.resize(h_angles_num);
+    for (int i = 0; i < h_angles_num; i++) {
+        _intensities[i].reserve(v_angles_num);
+        for (int j = 0; j < v_angles_num; j++) {
+            _intensities[i].push_back((float)(factor * parser.get_double()));
+        }
+    }
+
+    return !parser.has_error();
+}
+
+bool IESFile::process_type_b() {
+    std::vector<std::vector<float>> newintensity;
+    newintensity.resize(_v_angles.size());
+
+    for (int i = 0; i < _v_angles.size(); i++) {
+        newintensity[i].reserve(_h_angles.size());
+        for (int j = 0; j < _h_angles.size(); j++) {
+            newintensity[i].push_back(_intensities[j][i]);
+        }
+    }
+    _intensities.swap(newintensity);
+    _h_angles.swap(_v_angles);
+
+    float h_first = _h_angles[0], h_last = _h_angles[_h_angles.size() - 1];
```

Comment:
A common IES file type found in the wild has a single horizontal angle value, resulting in an h_last of 0.f. In this case, the cone angle will be less than 180 degrees. The emissive pattern is radially symmetric and can be constructed trivially. A 1d vector can also be interpolated as an optimization for the shader, but ... exercise for the reader.

Reply:
This is currently covered in the Type C handling, but not for B/A, since the standard only mentions the "single horizontal angle" case for Type C.
It wouldn't be hard to add it to B/A, though, and doesn't hurt to have.